src/pickup.cpp

```diff
@@ -214,7 +215,32 @@ bool pick_one_up( item_location &loc, int quantity, bool &got_water, bool &offer
         picked_up = true;
         option = NUM_ANSWERS; //Skip the options part
     } else if( newit.made_of_from_type( LIQUID ) ) {
-        got_water = true;
+        if( newit.has_flag( "FROZEN" ) ) {
+            if( u.has_quality( quality_id( "HAMMER" ), 1, 1 ) ) {
+                int units = newit.charges;
+                std::string action = query_popup()
+                                     .context( "YESNO" )
+                                     .message( _( "Do you want to gather %d %s?" ), units, newit.display_name() )
+                                     .option( "YES" )
+                                     .option( "NO" )
+                                     .cursor( 1 )
+                                     .query()
+                                     .action;
+                if( action == "YES" ) {
+                    option = STASH;
+                }
+            } else {
+                query_popup()
+                .context( "YES" )
+                .message( "%s", "You don't have a Hammering 1 tool." )
```

Comment:
Needs to be translatable - see "doc/TRANSLATING.md".

Reply:
Had a thorough look through the documentation and I think I fixed the faulty strings.